ExamenProeveUnity/Assets/Scripts/Runtime/Player/PlayerMovementHandler.cs

```diff
@@ -62,17 +63,22 @@ private void Update()
         /// <param name="setConstrained"></param>
         private void ConstrainPlayer(bool setConstrained)
         {
-            RigidbodyConstraints constraints = hip.constraints;
+            RigidbodyConstraints constraints = hipRigidbody.constraints;
             if (setConstrained)
             {
                 constraints = RigidbodyConstraints.FreezePosition;
             }
             else
             {
-                constraints = RigidbodyConstraints.FreezePositionY;
+                if (_horizontalMoveValue is not 0f && _verticalMoveValue is not 0f)
+                    constraints = RigidbodyConstraints.FreezePositionY;
+                else if (_horizontalMoveValue is not 0f)
+                    constraints = RigidbodyConstraints.FreezePositionZ | RigidbodyConstraints.FreezePositionY;
+                else if (_verticalMoveValue is not 0f)
+                    constraints = RigidbodyConstraints.FreezePositionX | RigidbodyConstraints.FreezePositionY;
```

Comment:
this would not work since "is" and "is not" is a type validation and not a value validation